### PR TITLE
Sham

### PR DESCRIPTION
**Pull request by kingsham**

### CODE REVIEW

File: Backend/models/Timeslots.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+        waitlisted_students: Array,
+        
+        current_capacity: Number,
+        
+        max_capacity:Number,
+
+        start_time: Date, String,
+
+        end_time: Date, String
```
Comment (mikeazz) [inline Backend/models/Timeslots.js:18]: Why String?

File: Backend/routes/timeslot.js [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+const router = express.Router();
+const mongoose = require('mongoose');
+const { db } = require('../models/Timeslots');
+mongoose.set('debug', true);
+const bodyParser = require('body-parser');
+const Timeslots = require('../models/Timeslots');
+const { exec } = require('child_process');
+​
+​
+//GET POSTS:
```
Comment (mikeazz) [inline Backend/routes/timeslot.js]: Timeslots, not posts (reoccurs in other comments)

File: Backend/routes/timeslot.js [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+        if (err) return res.json({ "message": err });
+        res.json(timeslots);
+    });
+​
+});
+​
+​
+​
+//DELETE ALL POST:
+router.delete('/:postID', async(req, res) => {
```
Comment (mikeazz) [inline Backend/routes/timeslot.js]: timeSlotID

File: Backend/routes/timeslot.js [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+    await Post.remove((err, timeslots) => {
```
Comment (mikeazz) [inline Backend/routes/timeslot.js]: Timeslots.remove

File: Backend/models/Waitlist.js [hunk 5]
```diff
@@ -0,0 +1,15 @@
+const mongoose = require('mongoose');
+​
+const WaitlistSchema = mongoose.Schema({
```
Comment (mikeazz) [inline Backend/models/Waitlist.js]: Is this schema still needed, now that Timeslots includes waitlisted and reserved students?

File: Backend/models/reservation-mod.js [hunk 6]
```diff
@@ -0,0 +1,20 @@
+const mongoose = require('mongoose');
+
+const ReservationSchema = mongoose.Schema({
```
Comment (mikeazz) [inline Backend/models/reservation-mod.js:3]: Where will you use this?

File: Backend/routes/waitlisted.js [hunk 7]
```diff
@@ -0,0 +1,87 @@
...
+​
+    await Timeslots.findOne({
+        start_time: timeslots.start_time
+    }).exec((err, timeslots) => {
+        if (err) {
+            return res.json({ "message": err });;
+        } else {
+            if (timeslots.current_capacity != timeslots.max_capacity) {
+                timeslots.reserved_students.push(req.body.reserved_students[0]);
+                timeslots.current_capacity = timeslots.current_capacity + 1;
```
Comment (mikeazz) [inline Backend/routes/waitlisted.js:47]: This is dangerous. If multiple writes increment the current_capacity in parallel, the value will be incorrect.

We don't need a current_capacity, we could count items in the database

File: Backend/routes/waitlisted.js [hunk 8]
```diff
@@ -0,0 +1,87 @@
...
+        "waitlisted_students": req.body.waitlisted_students
+    });
+​
+    await Timeslots.findOne({
+        start_time: timeslots.start_time
+    }).exec((err, timeslots) => {
+        if (err) {
+            return res.json({ "message": err });;
+        } else {
+            if (timeslots.current_capacity != timeslots.max_capacity) {
```
Comment (mikeazz) [inline Backend/routes/waitlisted.js]: you should check whether current_capacity < max_capacity

**Review by mikeazz — COMMENTED** [review]
Nice progress. I added a few questions/comments.

General comments
1. Please remove the `node_modules` from your commit. We should not include our dependencies to the codebase
2. Make sure to rebase your branch from master. The `Backend` directory no longer exists :)

**Comment by kingsham** [comment]
How do I rebase again ? I forgot

**Review by kingsham — COMMENTED** [review]
removed node modules

**Pull request merged**